python/ray/tests/mock_s3_server.py

```diff
@@ -0,0 +1,88 @@
+# extracted from aioboto3
```

Comment:
should we move the workflow one to here directly? (I'm ok with deleting that in another PR)

Reply:
yes, I will use this in workflow tests after this PR..